### ISSUE TITLE
Question: Does ADF support nodejs 14?

### ISSUE THREAD

**Issue by engrun** [awslabs/aws-deployment-framework#385]
Context

Trying to use node14 for deploying SAM app. 

buildspec.yml
```phases:
  install:
    runtime-versions:
      python: 3.8
      nodejs: 14
```

pipeline-definition: 
``` build:
        provider: codebuild
        properties:
          spec_filename: buildspec.yml
          image: STANDARD_5_0
```

When running the pipeline, this is the relevant logging: 

```
[Container] 2021/09/02 09:50:29 Running command bash adf.sh transform
302 | adf generate parameters...
..
.. an error and a warning related to pip
..
312 | adf package transform...
323 | Building codeuri: ./lib/handlers runtime: nodejs14.x metadata: {'aws:cdk:path': 
..
..
325 | Build Failed
326 | Error: 'nodejs14.x' runtime is not supported
```

*Question* 
Is this expected behaviour?  
Or is this a misconfiguration on my part?

**Comment by engrun** [comment]
To answer my own question: 
It turns out that ADF does not use the aws-sam-cli that comes with the CodeBuild image.

Resolution: 
When  changing 
`aws-sam-cli==1.15.0` to `aws-sam-cli==1.27.2`

in the file

`/aws-deployment-framework-bootstrap/adf-build/shared/requirements.txt`

then
`sam build` completes successfully

(if you want to test this you can change this file in the ADF shared modules bucket directly for a faster feedback-loop)

**Comment by sbkok** [comment]
Thanks @engrun for providing your workaround.

This issue will be resolved with the next release of ADF.
We are working on v3.2.0 that will update the SAM CLI too, so others don't need to manually change these requirements in the future.

I will leave this issue open until the next version of ADF is released.

**Comment by sbkok** [comment]
Thank you for your patience. I am happy to inform you that this issue has been [resolved in our latest release v3.2.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.2.0) just now.
I'm hereby closing this issue. Please open a new issue if you are experiencing any issues with the latest release.
